### PR TITLE
Add a get topic REST API

### LINKED ISSUE #774: HIP-874: Get topic REST Java implementation
**Problem**

Currently `/api/v1/topics/${id}/messages` exist to retrieve messages of a topic. However, there's no API to get the topic details.

**Solution**

Add an `/api/v1/topics/${id}` endpoint to expose a single topic's info.

```json
{
  "admin_key": {
      "_type": "ED25519",
      "key": "bfc9d4a344389324119eed485eb3a875b1e5a54bd189d2fd5ad88ce1d3473dee"
  },
  "auto_renew_account": "0.0.654321",
  "auto_renew_period": 7890000,
  "created_timestamp": "1707953087.386834512",
  "deleted": false,
  "memo": "Atma",
  "submit_key": {
      "_type": "ED25519",
      "key": "bfc9d4a344389324119eed485eb3a875b1e5a54bd189d2fd5ad88ce1d3473dee"
  },
  "timestamp": {
    "from": "1707953087.386834512",
    "to": null
  },
  "topic_id": "0.0.123456"
}
```

**Alternatives**

See #773 as there will be overlap

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add an `/api/v1/topics/{id}` REST API
* Add a `ControllerTest` hierarchy
* Normalize OpenAPI operationId naming

**Related issue(s)**:

Fixes #774

**Notes for reviewer**:

Did not apply `ControllerTest` to `AllowancesControllerTest` yet since other PRs are modifying it.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/GenericControllerAdvice.java [hunk 1]
```diff
@@ -95,6 +96,18 @@ private ResponseEntity<Error> genericError(final Exception e) {
...
 
+    @Override
+    @SuppressWarnings("unchecked")
+    protected ResponseEntity<Object> handleMissingPathVariable(
+            MissingPathVariableException ex, HttpHeaders headers, HttpStatusCode status, WebRequest request) {
+        if (ex.isMissingAfterConversion()) {
+            var message = "Invalid path variable '" + ex.getVariableName() + "'";
+            ResponseEntity<?> response = errorResponse(message, status);
+            return (ResponseEntity<Object>) response;
+        }
```
Comment (jascks) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/GenericControllerAdvice.java:107]: A `MissingPathVariableException` where `isMissingAfterConversion()` is true represents an unexpected/invalid path variable? I see in the super class `MissingRequestValueException`:
```java
	/**
	 * Whether the request value was present but converted to {@code null}, e.g. via
	 * {@code org.springframework.core.convert.support.IdToEntityConverter}.
	 */
	public boolean isMissingAfterConversion() {
		return this.missingAfterConversion;
	}
```
That sounds like it means the path variable was present but its value could not be converted. That's the takeaway from `TopicControllerTest.invalidId()` as well. Shouldn't the messaging should be little different, say:
```java
        var message = "Invalid value for path variable '" + ex.getVariableName() + "'";
```

Comment (steven-sheehy) [reply hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/GenericControllerAdvice.java:107]: Seems like you're thinking the messages implies an _unexpected_ path variable which I didn't get that impression, but I changed it to add `value for`.

Comment (jascks) [reply hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/controller/GenericControllerAdvice.java:107]: Thanks! Yes I did get that impression. :smile:

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java [hunk 2]
```diff
@@ -55,4 +61,10 @@ public EntityId lookup(EntityIdParameter accountId) {
 
         return id.orElseThrow(() -> new EntityNotFoundException("No account found for the given ID"));
     }
+
+    private void validateShard(Object id, long shard) {
+        if (shard != properties.getShard()) {
+            throw new IllegalArgumentException("ID %s has an invalid shard".formatted(id));
```
Comment (jascks) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java:67]: In light of DevRel/docs wanting to make things more self documenting etc., should we return what shard is acceptable?

Comment (steven-sheehy) [reply hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java]: Added that

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/ControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                    .retrieve()
+                    .toBodilessEntity()
+                    .getHeaders();
+
+            // Then
+            assertThat(headers.getAccessControlAllowOrigin()).isEqualTo("*");
+        }
+
+        @Test
+        void unsupportedMediaException() {
```
Comment (jascks) [inline hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/ControllerTest.java:111]: Should this be named `unsupportedMethodException()`?   Also, what if a controller's `defaultRequest()` is a POST?

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java [hunk 4]
```diff
@@ -64,7 +64,7 @@ public EntityId lookup(@NotNull EntityIdParameter accountId) {
 
     private void validateShard(Object id, long shard) {
         if (shard != properties.getShard()) {
-            throw new IllegalArgumentException("ID %s has an invalid shard".formatted(id));
+            throw new IllegalArgumentException("Unsupported shard %s in ID %s".formatted(shard, id));
```
Comment (jascks) [inline hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java:67]: Sorry, I was thinking of communicating on how to fix it, assuming they at least know `shard.realm.id`:
```java
throw new IllegalArgumentException("ID %s has an invalid shard. Shard must be %d".formatted(id, properties.getShard()));
```

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/TopicControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+            // When
+            ThrowingCallable callable = () -> restClient
+                    .get()
+                    .uri("", entity.toEntityId().toString())
+                    .retrieve()
+                    .body(Topic.class);
+
+            // Then
+            validateError(callable, HttpClientErrorException.NotFound.class, "not found");
```
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/TopicControllerTest.java]: added

File: hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/ControllerTest.java (hunk 3 above)
Comment (steven-sheehy) [reply hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/ControllerTest.java]: It was a copy from AllowancesControllerTest, but I can fix. We'll worry about POST when we cross that bridge.

File: hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java (hunk 4 above)
Comment (steven-sheehy) [reply hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/service/EntityServiceImpl.java]: Adjusted

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Pull request merged**